src/HtmlUtils.tsx

```diff
@@ -601,6 +603,28 @@ export function bodyToHtml(content: IContent, highlights: string[], opts: IOpts
         </span>;
 }
 
+/* turn a room topic into html
+ *
+ * topic: plain text topic
+ * htmlTopic: optional html topic
+ * ref: React ref to attach to any React components returned
+ */
+export function topicToHtml(topic: string, htmlTopic?: string, ref?: React.Ref<HTMLSpanElement>): ReactNode {
+    const html = SettingsStore.getValue("feature_html_topic") ? htmlTopic : null;
+    return bodyToHtml({
```

Comment:
This is maybe a question of whether we want to allow extended topics with _richer_ content below the second line as described [here](https://github.com/matrix-org/matrix-spec-proposals/pull/3765#discussion_r842011219). If that's pushing the limits too far, agree that extracting a custom function would be better.

Reply:
Looking into this some more, preventing the conversion of e.g. `# Heading` to `<h1>Heading</h1>` seems tricky with the CommonMark HTML renderer we're using. The conversion also happens when sending the `m.room.topic` event. This function here just sanitizes the HTML for display and performs a few other tweaks.

I pushed a fix that uses CSS to force headings in the timeline topic to the same size and style as normal text via CSS. In the `/topic` dialog, it seems ok to allow headings & friends given that there is more space available here?

Reply:
![Screenshot 2022-04-07 at 22 00 29](https://user-images.githubusercontent.com/1137962/162286333-09ff4d0c-8c84-40fa-b70d-25b67e356409.png)

Reply:
Err, my assumption was that we'd run the resulting HTML through a sanitizer (which is done internally in `bodyToHtml`) with different params. We'd still let the topic contain the HTML (because we can't stop that), but we'd sanitize it more aggressively on the receiving end.

Reply:
Gotcha. I've now extracted a separate function, stripped down the allowed tags and removed the processing of maths and other unrelated things.